kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -316,7 +316,7 @@ private[kyuubi] class EngineRef(
   def deregister(discoveryClient: DiscoveryClient, hostPort: (String, Int)): Unit =
     tryWithLock(discoveryClient) {
       if (discoveryClient.getServerHost(engineSpace) == Option(hostPort)) {
-        discoveryClient.delete(engineSpace)
+        discoveryClient.delete(engineSpace, deleteChildren = true)
```

Comment:
`deleteChildren = true` Is this required?

Reply:
I make a followup for this: https://github.com/apache/kyuubi/pull/5910